src/Obelisk/Frontend/GoogleAnalytics.hs

```diff
@@ -1,54 +1,250 @@
+instance HasDocument m => HasDocument (GoogleAnalyticsT t w m)
+
+instance TriggerEvent t m => TriggerEvent t (GoogleAnalyticsT t w m) where
+  newTriggerEvent = lift newTriggerEvent
+  newTriggerEventWithOnComplete = lift newTriggerEventWithOnComplete
+  newEventWithLazyTriggerWithOnComplete = lift . newEventWithLazyTriggerWithOnComplete
+
+instance HasJSContext m => HasJSContext (GoogleAnalyticsT t w m) where
+  type JSContextPhantom (GoogleAnalyticsT t w m) = JSContextPhantom m
+  askJSContext = lift askJSContext
+
+instance MonadSample t m => MonadSample t (GoogleAnalyticsT t w m) where
+  sample = lift . sample
+
+instance MonadHold t m => MonadHold t (GoogleAnalyticsT t w m) where
+  {-# INLINABLE hold #-}
+  hold v0 = lift . hold v0
+  {-# INLINABLE holdDyn #-}
+  holdDyn v0 = lift . holdDyn v0
+  {-# INLINABLE holdIncremental #-}
+  holdIncremental v0 = lift . holdIncremental v0
+  {-# INLINABLE buildDynamic #-}
+  buildDynamic a0 = lift . buildDynamic a0
+  {-# INLINABLE headE #-}
+  headE = lift . headE
+
+instance HasJS x m => HasJS x (GoogleAnalyticsT t w m) where
+  type JSX (GoogleAnalyticsT t w m) = JSX m
+  liftJS = lift . liftJS
+
+type TrackingId = Text
 
 -- | Config path for the tracking ID
 trackingIdPath :: Text
 trackingIdPath = "frontend/google/analytics/tracking-id"
 
-type TrackingId = Text
-
 -- | Embed Google analytics scripts. This function will throw if the tracking ID
 -- config is missing.
-googleAnalyticsFromConfig :: (HasConfigs m, Prerender js t m, PerformEvent t m, DomBuilder t m, Routed t r m) => m ()
-googleAnalyticsFromConfig = getTrackingId >>= \case
-   Nothing -> fail $ T.unpack $ T.unwords
-     [ "Google analytics tracking ID not found in configs."
-     , "Expected path: " <> trackingIdPath
-     ]
-   Just tid -> googleAnalytics tid
+googleAnalyticsFromConfig :: (HasConfigs m, Prerender js t m, DomBuilder t m, Routed t r m) => m ()
+googleAnalyticsFromConfig = getTrackingId >>= googleAnalytics
 
 -- | Embed Google analytics scripts with the given tracking ID.
-googleAnalytics :: (DomBuilder t m, Prerender js t m, PerformEvent t m, Routed t r m) => TrackingId -> m ()
-googleAnalytics trackingId = do
-  let gtagSrc = "https://www.googletagmanager.com/gtag/js?id=" <> trackingId
-      escapedTrackingId = T.pack $ show trackingId
-  elAttr "script" ("async" =: "async" <> "src" =: gtagSrc) blank
-  el "script" $ text $ T.unlines
-    [ "window.dataLayer = window.dataLayer || [];"
-    , "function gtag(){dataLayer.push(arguments);}"
-    , "gtag('js', new Date());"
-    , "gtag('config', " <> escapedTrackingId <> ");"
-    ]
-  route <- askRoute
-  prerender_ blank $ do
-    performEvent_ $ ffor (updated route) $ \_ -> liftJSM $ do
-      void $ eval $ "gtag('config', " <> escapedTrackingId <> ", { 'page_path': location.pathname, 'page_title': document.title });"
+googleAnalytics :: (DomBuilder t m, Prerender js t m, Routed t r m) => Maybe TrackingId -> m ()
```

Comment:
Why does this need `Routed`?

Reply:
Because the automatically tells Google Analytics which page that you are looking at;  which is based on a dynamic routing behavior.